x/gov/keeper/querier.go

```diff
@@ -177,14 +177,14 @@ func queryTally(ctx sdk.Context, path []string, req abci.RequestQuery, keeper Ke
 
 // nolint: unparam
 func queryVotes(ctx sdk.Context, path []string, req abci.RequestQuery, keeper Keeper) ([]byte, sdk.Error) {
-	var params types.QueryProposalParams
+	var params types.QueryProposalVotesParams
 	err := keeper.cdc.UnmarshalJSON(req.Data, &params)
 
 	if err != nil {
 		return nil, sdk.ErrUnknownRequest(sdk.AppendMsgToErr("incorrectly formatted request data", err.Error()))
 	}
 
-	votes := keeper.GetVotes(ctx, params.ProposalID)
```

Comment:
Keep this call and do the pagination call here instead. This is what other queriers do.

Reply:
tbh it makes more sense to paginate in keeper. current implementation always loads whole data set, but we can change to skip the required number of records without loading the body (if we want). and caller doesn't have to know anything about it

for example leveldb iterator allows it, i haven't checked if iavl kv wrapper makes it harder/different. but this separation of concerns makes sense, right?

Reply:
No, we should try and keep pagination as much as possible in the querier logic, not the keepers (I know this isn't always the case). To your point, the entire set is loaded regardless -- doesn't matter if its in the keeper or querier.

Reply:
> The only way to perform less IO is to have the iteration modified in the keepers and that would require pagination in the keepers as you've suggested, which would mean no calls to client.Paginate.

exactly, but this is a kepeer implementation detail. my point was that if later loading the whole data set turns out to be a problem - there will be no changes in querier

Reply:
True. Still, let's add it to the querier. Let's discuss in another issue together how to support this natively. e.g. `sdk.NewPrefixIteratorWithPagination(...)`